core/shared/src/main/scala/scalaz/zio/Exit.scala

```diff
@@ -20,7 +20,7 @@ package scalaz.zio
  * An `Exit[E, A]` describes the result of executing an `IO` value. The
  * result is either succeeded with a value `A`, or failed with a `Cause[E]`.
  */
-sealed abstract class Exit[+E, +A] extends Product with Serializable { self =>
+sealed trait Exit[+E, +A] extends Product with Serializable { self =>
```

Comment:
Interesting, why the change? I thought abstract classes generate more efficient bytecode most of the time.

Reply:
We dropped Scala 2.11. The main reason for abstract class is to avoid breaking backward binary compatibility on adding a method with a default implementation.